### PR TITLE
@fullstack-one/cli

### PR DESCRIPTION
**Pull request by davidsparkles**
- Added the new package `cli`
- Configured npm executables
- Added empty commands for initialization, migrating the database and generating typescript files.
- A bit unrelated but still: use .env properties in the examples `./config/default.js`

### CODE REVIEW

File: packages/cli/lib/generate-ts/dbMetaToTsMapper.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  return _.map(dbMetaTables, (table) => {
+    const constraints = table.constraints;
+    const tsProperties: string = _.map(table.columns, (column) => mapDbMetaColumnToTypescriptProperty(column, constraints, enums)).join("\n");
+    return `export interface I${pascal(table.name)} {\n${tsProperties}\n}`;
+  }).join("\n\n");
+}
+
+function mapDbMetaColumnToTypescriptProperty(dbMetaColumn: IColumn, constraints: { [name: string]: IConstraint }, enums: string[]): string {
+  const { name, type, constraintNames, customType, defaultValue } = dbMetaColumn;
+  const isMandatory = _.find(constraintNames, (constraintName) => constraints[constraintName].type === "NOT NULL") !== undefined;
```
Comment (terehov) [inline packages/cli/lib/generate-ts/dbMetaToTsMapper.ts]: find is natively support by ES.

File: packages/cli/lib/generate-ts/dbMetaToTsMapper.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  return _.map(dbMetaEnums, (enumItem) => {
+    const values: string = _.map(enumItem.values, (value) => `  ${pascal(value)} = "${value}"`).join(",\n");
+    return `export enum ${enumItem.name} {\n${values}\n}`;
+  }).join("\n\n");
+}
+
+function mapDbMetaTablesToTypescriptInterfaces(dbMetaTables: { [name: string]: ITable }, enums: string[] = []): string {
+  return _.map(dbMetaTables, (table) => {
+    const constraints = table.constraints;
+    const tsProperties: string = _.map(table.columns, (column) => mapDbMetaColumnToTypescriptProperty(column, constraints, enums)).join("\n");
```
Comment (terehov) [inline packages/cli/lib/generate-ts/dbMetaToTsMapper.ts]: map is nativelly supportet by ES: 
e.g.
`Object.values(table.columns).map(column => {...})`

File: packages/cli/lib/generate-ts/dbMetaToTsMapper.ts (hunk 1 above)
Comment (terehov) [reply packages/cli/lib/generate-ts/dbMetaToTsMapper.ts]: `Object.values(constraintNames).find(...)`

File: packages/cli/lib/generate-ts/dbMetaToTsMapper.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    int4: "number",
+    float8: "number",
+    bool: "boolean",
+    uuid: "string",
+    jsonb: "any",
+    relation: "any",
+    enum: "any",
+    customType: "any"
+  };
+  if (columnType === "enum" && _.includes(enums, customType)) return customType;
```
Comment (terehov) [inline packages/cli/lib/generate-ts/dbMetaToTsMapper.ts]: includes is also part of ES.

File: packages/cli/lib/generate-ts/dbMetaToTsMapper.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    return `export interface I${pascal(table.name)} {\n${tsProperties}\n}`;
+  }).join("\n\n");
+}
+
+function mapDbMetaColumnToTypescriptProperty(dbMetaColumn: IColumn, constraints: { [name: string]: IConstraint }, enums: string[]): string {
+  const { name, type, constraintNames, customType, defaultValue } = dbMetaColumn;
+  const isMandatory = _.find(constraintNames, (constraintName) => constraints[constraintName].type === "NOT NULL") !== undefined;
+  const property: string = `${name}${isMandatory ? "" : "?"}`;
+  const tsType: string = dbMetaColumnTypeToTypescriptType(type, customType, enums);
+  const comment = [type, `${customType || ""}`, `${defaultValue ? defaultValue.value || "" : ""}`].filter((item) => item !== "").join(" ");
```
Comment (terehov) [inline packages/cli/lib/generate-ts/dbMetaToTsMapper.ts]: u see, u already use `filter`, `find` is essentially the same, but returns the first found element (not an array)

File: packages/cli/lib/generate-ts/index.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+console.log(`Assumed project main file: ${projectMainFile}`);
+console.log();
+
+const dotEnvPath = `${currentWorkDirectory}/.env`;
+dotenv.config({ path: dotEnvPath });
+// Manipulate main file name, so @fullstack-one/config gets the config and we can get the env.
+require.main.filename = projectMainFile;
+
+console.log("Boot ...");
+Container.get(Auth);
```
Comment (terehov) [inline packages/cli/lib/generate-ts/index.ts]: We really need to find a better solution for that. One could be, that the packages manipulating GQl would need to register somewhere.

File: packages/cli/package.json [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    "@fullstack-one/boot-loader": "^0.4.56",
+    "@fullstack-one/di": "^0.4.56",
+    "@fullstack-one/file-storage": "^0.4.56",
+    "@fullstack-one/schema-builder": "^0.4.56",
+    "@types/graphql": "^0.13.2",
+    "change-case": "^3.1.0",
+    "commander": "^2.19.0",
+    "dotenv": "^6.2.0",
+    "graphql": "^0.13.2",
+    "graphql-code-generator": "^0.16.0",
```
Comment (terehov) [inline packages/cli/package.json]: do we still need this dependencies?

File: packages/cli/lib/generate-ts/index.ts (hunk 5 above)
Comment (davidsparkles) [reply packages/cli/lib/generate-ts/index.ts]: We should attack this issue in a seperate PR as it requires changes in the `config` package.

**Pull request merged**